### PR TITLE
HHH-12921 HHH-12922 Hibernate Connection Pool Thread Improvements

### PR DESCRIPTION
**Pull request by caristu**
Improvements for Hibernate Connection Pool Validation Thread:

- Set the thread as a daemon thread
- Give it a name

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java [hunk 1]
```diff
@@ -453,4 +454,13 @@ public void closeConnection(Connection conn) throws SQLException {
 		}
 	}
 
+	private class ValidationThreadFactory implements ThreadFactory {
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java]: I think you are missing some imports (at least for `ThreadFactory`) here.

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java [hunk 2]
```diff
@@ -384,7 +384,8 @@ private void startIfNeeded() {
 				if ( active ) {
 					return;
 				}
-				executorService = Executors.newSingleThreadScheduledExecutor();
+				ValidationThreadFactory vtf = new ValidationThreadFactory();
+				executorService = Executors.newSingleThreadScheduledExecutor(vtf);
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java]: A reminder for me: we need to reformat the new code to comply with the Hibernate formatting rules.

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java (hunk 1 above)
Comment (vladmihalcea) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java]: @gsmet I've added those when reviewing it and want to add a test for it as well. I'll push a PR to replicate this one when I'm done.

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java (hunk 2 above)
Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java]: @vladmihalcea can you also reformat the added code properly too if you're working on this PR? Thanks!

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java (hunk 1 above)
Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java]: FWIW, I'm not sure we'll be able to implement a test for that. If we don't, it's not a biggie IMHO.

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java (hunk 2 above)
Comment (vladmihalcea) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java]: @gsmet natürlich.

**Comment by gsmet** [comment]
@caristu good catch! Nobody should use this in production but I agree it's better to get it right.

**Comment by caristu** [comment]
Hello @gsmet, @vladmihalcea  
Thanks for taking a look in so short time...I've just pushed two changesets I missed when preparing the PR...sorry for that

**Comment by vladmihalcea** [comment]
@caristu @gsmet I created [this PR](https://github.com/hibernate/hibernate-orm/pull/2498) after reviewing this one. Let me know what you think.

**Comment by gsmet** [comment]
Closed in favor of https://github.com/hibernate/hibernate-orm/pull/2498 .

**Pull request closed without merging**